templates/vue/src/components/SettingsModal.vue

```diff
@@ -98,23 +193,84 @@ export default {
       this.backgroundUrl = backgroundUrl
       this.autoLayout = autoLayout
       this.nodeDraggable = nodeDraggable
+      this.defaultPermissions = defaultPermissions
+      this.showAccess = showAccess
     },
     async updateSettings() {
       const settings = Object.assign(this.settings, {
         backgroundUrl: this.backgroundUrl,
         autoLayout: this.autoLayout,
         nodeDraggable: this.nodeDraggable,
+        defaultPermissions: this.defaultPermissions,
+        showAccess: this.showAccess,
       })
       await this.$store.dispatch("updateSettings", settings)
       // TODO: Improve behavior so refresh is not required (currently auto-layout and setting the background image only happen initially)
       // this.$emit("settings-updated", settings);
       // this.closeModal();
       location.reload()
     },
+    updatePermissions(value, rowName, type) {
+      if (rowName.startsWith("user") || wpData.roles.hasOwnProperty(rowName)) {
+        return this.changeIndividualPermission(value, rowName, type)
+      }
+      const rowIndex = Helpers.getPermissionRowIndex(rowName, this.permissionsOrder)
+      const lowerPriorityPermissions = this.permissionsOrder.slice(rowIndex + 1)
+      lowerPriorityPermissions.forEach(newRow => {
+        this.changeIndividualPermission(value, newRow, type)
+      })
+    },
+    isPermissionDisabled(rowName, type) {
+      if (rowName == "public") {
+        return false
+      }
+      // keep going up until we find a non-user higher row
+      const rowIndex = Helpers.getPermissionRowIndex(rowName, this.permissionsOrder)
+      const higherRow = this.permissionsOrder[rowIndex - 1]
+      if (higherRow.startsWith("user") || wpData.roles.hasOwnProperty(higherRow)) {
+        return this.isPermissionDisabled(higherRow, type)
+      }
+      const permissions = this.defaultPermissions[higherRow]
+      if (permissions) {
+        return permissions.includes(type)
+      }
+      return false
+    },
+    changeIndividualPermission(value, rowName, type) {
+      let currentPermissions = this.defaultPermissions[rowName]
+      if (!currentPermissions) {
+        currentPermissions = []
+      }
+      let newPermissions = [...currentPermissions]
+      if (value) {
+        if (!currentPermissions.includes(value)) {
+          newPermissions.push(value)
+        }
+      } else {
+        newPermissions = currentPermissions.filter(permission => permission !== type)
+      }
+      this.defaultPermissions[rowName] = newPermissions
+    },
+    addUserPermissionRow() {
+      const userId = this.userId
+      if (
+        userId &&
+        Helpers.onlyContainsDigits(userId) &&
+        $("#user-" + userId + "-editcell").val() != ""
+      ) {
+        this.$set(this.node.permissions, `user-${userId}`, this.newPermissions)
+        this.permissionsOrder.push(`user-${userId}`)
+        this.userId = null
+      } else {
+        alert("Enter valid user id")
+      }
+    },
```

Comment:
I believe all these functions I have highlighted here can be moved to the Helper. If it's possible to create a component here and use it in both places, that would be even better so that we don't have to use a helper function.